traffic/data/adsb/raw_data.py

```diff
@@ -0,0 +1,165 @@
+            pos_tmp.tolist(),
+            columns=["latitude", "longitude"],
+            index=pos_tmp.index,
+        )
+        os_pos.drop(["inv", "inv_time", "oe_flag"], axis=1, inplace=True)
+        result = pd.concat([os_pos, pos], axis=1, join="inner")
+        return result
+
+    @staticmethod
+    def velocity_extraction(os_vel) -> pd.DataFrame:
+        if os_vel.empty:
+            return os_vel
+        if os_vel["mintime"].dtypes == np.float64:
+            os_vel["mintime"] = pd.to_datetime(os_vel["mintime"], unit="s")
+        os_vel["speed_temp"] = os_vel["rawmsg"].apply(adsb.velocity)
+        os_vel[["speed", "heading", "vertical_rate", "type"]] = pd.DataFrame(
+            os_vel["speed_temp"].tolist(), index=os_vel.index
+        )
+        os_vel.drop(["speed_temp", "type"], axis=1, inplace=True)
+        return os_vel
+
+    @staticmethod
+    def identification_extraction(os_ide) -> pd.DataFrame:
+        if os_ide.empty:
+            return os_ide
+
+        if os_ide["mintime"].dtypes == np.float64:
+            os_ide["mintime"] = pd.to_datetime(os_ide["mintime"], unit="s")
+        os_ide["callsign"] = os_ide["rawmsg"].apply(callsign)
+        return os_ide
+
+    def feature_extraction(self, lat_ref=None, lon_ref=None) -> "RawData":
+        if "msg_type" not in self.data.columns:
+            self.get_type()
+
+        ide = RawData.identification_extraction(
+            self.data.loc[self.data["msg_type"] == 1]
+        )
+        vel = RawData.velocity_extraction(
+            self.data.loc[self.data["msg_type"] == 4]
+        )
+        pos = RawData.position_extraction(
+            self.data.loc[self.data["msg_type"] == 3], lat_ref, lon_ref
+        )
+        result = pd.concat(
+            [ide, vel, pos], axis=0, sort=False, ignore_index=True
+        )
+        result = result.sort_values(by=["mintime", "icao24"])
+        return RawData(result)
+
+    def get_type(self, inplace=False):
+        def get_typecode(msg):
+            tc = adsb.typecode(msg)
+            if 9 <= tc <= 18:
+                return 3
+            elif tc == 19:
+                return 4
+            elif 1 <= tc <= 4:
+                return 1
+            else:
+                return np.nan
+
+        if inplace:
+            self.data["msg_type"] = self.data.rawmsg.apply(get_typecode)
+            self.data.dropna(subset=["msg_type"], inplace=True)
+        else:
+            return self.data.rawmsg.apply(get_typecode)
+
+    def to_beast(self, time_fmt: str = "dump1090") -> pd.Series:
+        df_beast = self.data[["mintime", "rawmsg"]].copy()
+        if isinstance(df_beast.mintime.iloc[0], pd.datetime):
+            df_beast.mintime = df_beast.mintime.astype(np.int64) / 10 ** 9
+        encoder = encode_time.get(time_fmt, encode_time_dump1090)
+        df_beast["time"] = encoder(df_beast.mintime)
+        df_beast["message"] = "@" + df_beast.time + df_beast.rawmsg
+        return df_beast["message"]
```

Comment:
Ici je renverrai bien un RawData du coup, avec a minima `timestamp` (`mintime`) et `message` (c'est toujours relou d'avoir à redécoder le timestamp pour savoir où on en est non?).
On peut extraire la colonne `message` côté utilisateur non?

Reply:
La méthode, je l'ai créer pour renvoyer un dataframe qui est prêt à être exporter en fichier .beast. Mon cas d'utilisation est le suivant : 

df_beast = df.to_beast()
df_beast.to_csv('data.beast', header=False, index=False)

Le fichier data.beast est le même que ce qu'une antenne beast pourrait créer. Si l'utilisateur veut uniquement le message ou le timestamp, il peut utiliser le RawData qu'il a à la base (dans mon exemple df)

Reply:
```python
df_beast = df.to_beast()
# tu pourrais faire
df_beast['message'].to_csv('data.beast', header=False, index=False)
# ou 
df_beast.before("2019-12-25 09:00")['message'].to_csv('data.beast', header=False, index=False)
```

Reply:
Bon j'ai fait `assign_beast` and `to_beast` du coup; comme ça tout le monde est content!